README.md

<!-- README.md is generated from README.Rmd. Please edit that file -->

# CawthronColours <img src="man/figures/hex_logo.png" align="right" height="150"/>

<!-- badges: start -->
<!-- badges: end -->

## package in development - could break at any moment ….

The goal of `CawthronColours` is to to provide official Cawthron colour
palettes for use in R. The colour palettes are a mix of well known
(e.g. `viridis`) and custom palettes. All are colour blind friendly.

Inspiration for package implementation came from the the
[`Manu`](https://github.com/G-Thomson/Manu) colour palette package. All
palettes except `viridis` were designed by XXX.

## Installation

The package is currently hosted on GitHub and can be installed as
follows:

``` r
# install.packages("devtools")
devtools::install_github("cawthron/CawthronColours")
```

## Usage

The colour palettes are stored as a list named `cawthron_palettes`.
There are categorical (`caw_cat_*`), sequential (`caw_seq_*`) and
diverging (`caw_div_*`) palettes.You can see a list of the available
palettes like so:

``` r
library(CawthronColours)
names(cawthron_palettes)
#>  [1] "caw_div_1" "caw_div_2" "caw_div_3" "caw_seq_1" "caw_seq_2" "caw_seq_3"
#>  [7] "caw_seq_4" "caw_seq_5" "caw_cat_1" "caw_cat_2"
```

A helper function `get_pal()` returns the desired colour palette as a
vector: For example:

``` r
get_pal("caw_cat_1")
#>  [1] "#f48521" "#fbd872" "#acd58e" "#76c044" "#4a7637" "#2f4926" "#4ba791"
#>  [8] "#2f725e" "#004f52" "#175c7d" "#029cbd" "#7ac4d3" "#9b99cd" "#6860a0"
#> [15] "#702365" "#e8b5d4" "#85243f" "#c22c43" "#d45f74" "#f8d3ca"
```

There is also a helper function `print_pal()` which displays the palette
in the graphics window.

``` r
caw_col <- get_pal("caw_cat_1")
print_pal(caw_col)
```

![](man/figures/README-unnamed-chunk-4-1.png)<!-- -->

Below area few examples of plots made using
[`ggplot2`](https://ggplot2.tidyverse.org/). We use the
[`gapminder`](https://github.com/jennybc/gapminder) package to get some
toy data and
[`rnaturalearth`](https://github.com/ropensci/rnaturalearth) to get a
map of New Zealand.

``` r
library(gapminder)
library(ggplot2)
library(dplyr)

# For categorical colours we use scale_colour_manual (or scale_fill_manual)
ggplot(gapminder, aes(x = continent, y = lifeExp, colour = continent)) +
  geom_jitter(position = position_jitter(width = 0.1, height = 0)) +
  scale_colour_manual(values = get_pal("caw_cat_1")) +
  theme_classic() +
  theme(legend.position = "none")
```

![](man/figures/README-unnamed-chunk-5-1.png)<!-- -->

``` r
# for continuous or sequential palettes we use scale_fill_gradientn (or scale_colour_gradientn)
ggplot(faithfuld) +
  geom_tile(aes(waiting, eruptions, fill = density)) +
  scale_fill_gradientn(colours = get_pal("caw_seq_1"))  +
  theme_classic() +
  theme(legend.position = "none")
```

![](man/figures/README-unnamed-chunk-5-2.png)<!-- -->

``` r
# download a map of NZ
nz_map <- rnaturalearth::ne_states(country = "new zealand", returnclass = "sf")

# remove outlying islands
nz_map <- nz_map |>
  dplyr::filter(region %in% c("South Island", "North Island"))

# add a random variable for each region
nz_map <- nz_map %>%
  dplyr::select(name) |>
  dplyr::mutate(count = runif(dplyr::n(), 0, 100))

# plot - use scale_fill_gradientn and scale_colour_gradientn to colour the
# regions by the random variable we just created
ggplot(nz_map, aes(fill = count, colour = count)) +
  geom_sf() +
  scale_fill_gradientn(colours = get_pal("caw_seq_1"))  +
  scale_colour_gradientn(colours = get_pal("caw_seq_1"))  +
  theme_void() +
  theme(legend.position = "none")
```

![](man/figures/README-unnamed-chunk-5-3.png)<!-- -->

# Palettes

Overview of the different colour palettes

## Categorical palettes

### Cawthron rainbow - `caw_cat_1`

![](man/figures/README-unnamed-chunk-6-1.png)<!-- -->

### Warm and cool - `caw_cat_2`

![](man/figures/README-unnamed-chunk-7-1.png)<!-- -->

## Sequential palettes

### Tuaone - `caw_seq_1`

![](man/figures/README-unnamed-chunk-8-1.png)<!-- -->

### Viridis - `caw_seq_2`

![](man/figures/README-unnamed-chunk-9-1.png)<!-- -->

### Tāne Mahuta - `caw_seq_3`

![](man/figures/README-unnamed-chunk-10-1.png)<!-- -->

### Tangaroa - `caw_seq_4`

![](man/figures/README-unnamed-chunk-11-1.png)<!-- -->

### Papatūānuku - `caw_seq_5`

![](man/figures/README-unnamed-chunk-12-1.png)<!-- -->

## diverging palettes

### Sun to sea - `caw_div_1`

![](man/figures/README-unnamed-chunk-13-1.png)<!-- -->

### Hot to cold - `caw_div_2`

![](man/figures/README-unnamed-chunk-14-1.png)<!-- -->

### Whenua ki te wai - `caw_div_3`

![](man/figures/README-unnamed-chunk-15-1.png)<!-- -->


NEWS.md
# CawthronColours (development version)

* Initial CRAN submission.


R/col_pals.R

#------Define colour palettes: DIVERGING ------#

caw.dvg.1 <- c("#85243f", "#c22c43", "#ec7d49", "#fed087", "#F1e6b2", "#f4f2e4", "#c6e7df", "#80ccb8", "#54b598", "#007377", "#004851")
caw.dvg.2 <- c("#8d2d1f", "#ba5000", "#ef9033", "#e0b27c", "#f2e3c2", "#f8fcff", "#c6dce8", "#77c0e5", "#1e82c6", "#2b5a89", "#163e5e")
caw.dvg.3 <- c("#225733", "#228848", "#6cc24a", "#a4cf60", "#d1debb", "#f4f4e6", "#bfe6ec", "#5ca5db", "#047ab0", "#245389", "#203f6b")

#----- Make palettes function -----#
col.dvg1 <- colorRampPalette(caw.dvg.1)
col.dvg2 <- colorRampPalette(caw.dvg.2)
col.dvg3 <- colorRampPalette(caw.dvg.3)

#---------------------------------------------#
#---------------------------------------------#

#------Define colour palettes: SEQUENTIAL ------#

caw.seq.1 <- c("#f9f7eb", "#D9E1E2", "#BFCEC2", "#8cB59e", "#67a392", "#54827a", "#447873", "#1e5b5c", "#133f3d", "#0d3331")
caw.seq.2 <- c("#fbe515", "#abd037", "#6cbe49", "#24a26c", "#1d8778", "#226a7b", "#284f7a", "#303575", "#372464", "#330a40")
caw.seq.3 <- c("#f9f39d", "#ede655", "#c3d142", "#99bc42", "#4bad48", "#359646", "#2a7d3e", "#1a5e30", "#1e4425", "#093322")
caw.seq.4 <- c("#eff8fd", "#c7dced", "#a0c1de", "#77a6d0", "#4f8bc0", "#2771b0", "#0056a0", "#10497e", "#0e3e63", "#073450")
caw.seq.5 <- c("#FDEAD3", "#FDDAAD", "#FDD086", "#FFAA4D", "#FF8F1C", "#DC6B2F", "#E04E39", "#9E2A2B", "#76232F", "#3F2021")

#----- Make palettes function -----#
col.seq1 <- colorRampPalette(caw.seq.1)
col.seq2 <- colorRampPalette(caw.seq.2)
col.seq3 <- colorRampPalette(caw.seq.3)
col.seq4 <- colorRampPalette(caw.seq.4)
col.seq5 <- colorRampPalette(caw.seq.5)

#---------------------------------------------#
#---------------------------------------------#

#------Define colour palettes: Categorical 1 ------#

caw.adj.6 <- c("#f48521", "#fbd872", "#acd58e", "#4ba791", "#175c7d", "#7ac4d3")
caw.adj.8 <- c("#f48521", "#fbd872", "#acd58e", "#4a7637", "#4ba791", "#004f52", "#175c7d", "#7ac4d3")
caw.adj.10 <- c("#f48521", "#fbd872", "#acd58e", "#76c044", "#4a7637", "#4ba791", "#004f52", "#175c7d", "#7ac4d3", "#6860a0")
caw.adj.12 <- c("#f48521", "#fbd872", "#acd58e", "#76c044", "#4a7637", "#2f4926", "#4ba791", "#004f52", "#175c7d", "#7ac4d3", "#6860a0", "#f8d3ca")
caw.adj.14 <- c("#f48521", "#fbd872", "#acd58e", "#76c044", "#4a7637", "#2f4926", "#4ba791", "#2f725e", "#004f52", "#175c7d", "#7ac4d3", "#6860a0", "#d45f74", "#f8d3ca")
caw.adj.16 <- c("#f48521", "#fbd872", "#acd58e", "#76c044", "#4a7637", "#2f4926", "#4ba791", "#2f725e", "#004f52", "#175c7d", "#029cbd", "#7ac4d3", "#6860a0", "#85243f", "#d45f74", "#f8d3ca")
caw.adj.18 <- c("#f48521", "#fbd872", "#acd58e", "#76c044", "#4a7637", "#2f4926", "#4ba791", "#2f725e", "#004f52", "#175c7d", "#029cbd", "#7ac4d3", "#9b99cd", "#6860a0", "#702365", "#85243f", "#d45f74", "#f8d3ca")
caw.adj.20 <- c("#f48521", "#fbd872", "#acd58e", "#76c044", "#4a7637", "#2f4926", "#4ba791", "#2f725e", "#004f52", "#175c7d", "#029cbd", "#7ac4d3", "#9b99cd", "#6860a0", "#702365", "#e8b5d4", "#85243f", "#c22c43", "#d45f74", "#f8d3ca")

#------Define colour palettes: Categorical 2 ------#

caw.fresh.6 <- c("#7f1424", "#fbb15d", "#c45b28", "#9ecce5", "#3398d2", "#175c7d")
caw.fresh.8 <- c("#7f1424", "#ffcc28", "#fbb15d", "#c45b28", "#9ecce5", "#3398d2", "#175c7d", "#35356d")
caw.fresh.10 <- c("#7f1424", "#bc3635", "#ffcc28", "#fbb15d", "#c45b28", "#9ecce5", "#3398d2", "#00549e", "#175c7d", "#35356d")
caw.fresh.12 <- c("#7f1424", "#bc3635", "#ffcc28", "#fbb15d", "#ffe0ae", "#c45b28", "#9ecce5", "#3398d2", "#00549e", "#175c7d", "#35356d", "#c3d3c2")
caw.fresh.14 <- c("#7f1424", "#bc3635", "#ffcc28", "#fbb15d", "#ffe0ae", "#c45b28", "#9ecce5", "#3398d2", "#00549e", "#175c7d", "#d383b7", "#35356d", "#c3d3c2", "#205128")
caw.fresh.16 <- c("#7f1424", "#bc3635", "#ffcc28", "#fbb15d", "#ffe0ae", "#c45b28", "#9ecce5", "#3398d2", "#00549e", "#175c7d", "#d383b7", "#35356d", "#c3d3c2", "#205128", "#c3ce58", "#083631")
caw.fresh.18 <- c("#7f1424", "#bc3635", "#ffcc28", "#fbb15d", "#ffe0ae", "#c45b28", "#9ecce5", "#3398d2", "#00549e", "#175c7d", "#d383b7", "#a184bd", "#35356d", "#c3d3c2", "#75a54a", "#205128", "#c3ce58", "#083631")
caw.fresh.20 <- c("#7f1424", "#bc3635", "#ffcc28", "#fbb15d", "#ffe0ae", "#c45b28", "#9ecce5", "#3398d2", "#00549e", "#175c7d", "#d383b7", "#a184bd", "#35356d", "#c3d3c2", "#75a54a", "#205128", "#67bd45", "#c3ce58", "#737f7e", "#083631")


#----- Make palettes function -----#
col.adj6 <- colorRampPalette(caw.adj.6)
col.adj8 <- colorRampPalette(caw.adj.8)
col.adj10 <- colorRampPalette(caw.adj.10)
col.adj12 <- colorRampPalette(caw.adj.12)
col.adj14 <- colorRampPalette(caw.adj.14)
col.adj16 <- colorRampPalette(caw.adj.16)
col.adj18 <- colorRampPalette(caw.adj.18) # not yet working
col.adj20 <- colorRampPalette(caw.adj.20) # not yet working


col.fresh6 <- colorRampPalette(caw.fresh.6)
col.fresh8 <- colorRampPalette(caw.fresh.8)
col.fresh10 <- colorRampPalette(caw.fresh.10)
col.fresh12 <- colorRampPalette(caw.fresh.12)
col.fresh14 <- colorRampPalette(caw.fresh.14)
col.fresh16 <- colorRampPalette(caw.fresh.16)
col.fresh18 <- colorRampPalette(caw.fresh.18)
col.fresh20 <- colorRampPalette(caw.fresh.20)

#---------------------------------------------#
#---------------------------------------------#

R/col_functions.R
#' List of available palettes
#'
#' Official Cawthron colour palettes. Use \code{\link{get_pal}} to extract palettes.
#'
#' Palettes include:
#'
#'  * categorical: \code{caw_cat_1}, \code{caw_cat_2}
#'  * diverging: \code{caw_div_1}, \code{caw_div_2}, \code{caw_div_3}
#'  * sequential: \code{caw_seq_1}, \code{caw_seq_2}, \code{caw_seq_3}, \code{caw_seq_4}, \code{caw_seq_5}
#'
#' @export
cawthron_palettes <- list(
  #------Define colour palettes: DIVERGING ------#
  caw_div_1 = c("#85243f", "#c22c43", "#ec7d49", "#fed087", "#F1e6b2", "#f4f2e4", "#c6e7df", "#80ccb8", "#54b598", "#007377", "#004851"),
  caw_div_2 = c("#8d2d1f", "#ba5000", "#ef9033", "#e0b27c", "#f2e3c2", "#f8fcff", "#c6dce8", "#77c0e5", "#1e82c6", "#2b5a89", "#163e5e"),
  caw_div_3 = c("#225733", "#228848", "#6cc24a", "#a4cf60", "#d1debb", "#f4f4e6", "#bfe6ec", "#5ca5db", "#047ab0", "#245389", "#203f6b"),

  #------Define colour palettes: SEQUENTIAL ------#
  caw_seq_1 = c("#f9f7eb", "#D9E1E2", "#BFCEC2", "#8cB59e", "#67a392", "#54827a", "#447873", "#1e5b5c", "#133f3d", "#0d3331"),
  caw_seq_2 = c("#fbe515", "#abd037", "#6cbe49", "#24a26c", "#1d8778", "#226a7b", "#284f7a", "#303575", "#372464", "#330a40"),
  caw_seq_3 = c("#f9f39d", "#ede655", "#c3d142", "#99bc42", "#4bad48", "#359646", "#2a7d3e", "#1a5e30", "#1e4425", "#093322"),
  caw_seq_4 = c("#eff8fd", "#c7dced", "#a0c1de", "#77a6d0", "#4f8bc0", "#2771b0", "#0056a0", "#10497e", "#0e3e63", "#073450"),
  caw_seq_5 = c("#FDEAD3", "#FDDAAD", "#FDD086", "#FFAA4D", "#FF8F1C", "#DC6B2F", "#E04E39", "#9E2A2B", "#76232F", "#3F2021"),

  #------Define colour palettes: Categorical 1 ------#
  caw_cat_1 = c("#f48521", "#fbd872", "#acd58e", "#76c044", "#4a7637", "#2f4926", "#4ba791", "#2f725e", "#004f52", "#175c7d", "#029cbd", "#7ac4d3", "#9b99cd", "#6860a0", "#702365", "#e8b5d4", "#85243f", "#c22c43", "#d45f74", "#f8d3ca"),

  #------Define colour palettes: Categorical 2 ------#
  caw_cat_2 = c("#7f1424", "#bc3635", "#ffcc28", "#fbb15d", "#ffe0ae", "#c45b28", "#9ecce5", "#3398d2", "#00549e", "#175c7d", "#d383b7", "#a184bd", "#35356d", "#c3d3c2", "#75a54a", "#205128", "#67bd45", "#c3ce58", "#737f7e", "#083631")
)




#' Select Cawthron palette
#'
#' Return official Cawthron colour palettes.
#'
#' @param name Name of desired palette. Choices are:
#'
#'  * categorical: \code{caw_cat_1}, \code{caw_cat_2}
#'  * diverging: \code{caw_div_1}, \code{caw_div_2}, \code{caw_div_3}
#'  * sequential: \code{caw_seq_1}, \code{caw_seq_2}, \code{caw_seq_3}, \code{caw_seq_4}, \code{caw_seq_5}
#'
#' @return A vector of colours.
#' @export
#' @examples
#' get_pal("caw_div_1")
get_pal <- function(name) {

  pal <- cawthron_palettes[[name]]
  if (is.null(pal))
    stop("Palette not found.")

  pal
}

#' Plots a colour palette
#'
#' Visually inspect the given colour palette.
#'
#' @param col_pal a vector of colours
#' @export
#' @importFrom graphics par image
#' @examples
#' my_col_pal <- get_pal("caw_div_3")
#' print_pal(my_col_pal)
print_pal <- function(col_pal) {
  n <- length(col_pal)
  old <- par(mar = c(0.5, 0.5, 0.5, 0.5))
  on.exit(graphics::par(old))

  graphics::image(1:n, 1, as.matrix(1:n), col = col_pal,
        ylab = "", xaxt = "n", yaxt = "n", bty = "n")
}


#' Continuous colour palettes
#'
#'  Generate a continuous colour palette based on one of the existing colour palettes.
#'
#' @param pal name of the colour palette to extract two gradient end points from.
#' @param gradient_n number of colours in the continuous gradient
#'
#' @return a vector of colours
#' @importFrom grDevices colorRampPalette
#' @export
#' @examples
#' my_con_pal <- continuous_pal(pal = "caw_div_3", gradient_n = 100)
#' print_pal(my_con_pal)
continuous_pal <- function(pal = "caw_div_3", gradient_n = 100) {
  selected_colours <- get_pal(pal)
  grDevices::colorRampPalette(selected_colours)(gradient_n)
}